src/web/src/app/space/service.ts

```diff
@@ -123,230 +32,148 @@ export class SpaceService {
-			if (space) {
-				return of(space)
-					.pipe(
-						delay(2000),
-						src => this.requestWatcher.WatchFor(src)
-					);
-			}
-		}
+	getByLink(linkId: string): Observable<any> {
+		return this.http.get<Space>(this._apiUrl + "?link=" + linkId).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		);
 	}
 
-	getInvitationList(limit: number, page: number, spaceId: string = null): Observable<PaginatorApi<SpaceInvitation>> {
-		const grouped = spaceId ?
-			this.invitations.filter(item => item.spaceId === spaceId) :
-			this.invitations;
-
-		const res: PaginatorApi<SpaceInvitation> = {
-			count: grouped.length,
-			results: grouped.slice(limit * page, limit * page + limit),
-		};
-		return of(res)
-			.pipe(delay(2000));
+	getInvitationList(limit: number, page: number, spaceId: string = null): Observable<Paging<SpaceInvitation>> {
+		let url = this._apiUrl + "/invitation?start=" + limit*page + "&limit=" + limit;
+		if (spaceId)
+			url += '&spaceId=' + spaceId;
+		return this.http.get<Paging<SpaceInvitation>>(url).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		);
 	}
 
-	getUserList(spaceId: string, limit: number, page: number): Observable<PaginatorApi<SpaceUser>> {
-		let grouped: SpaceUser[] = [];
-		this.users.forEach(function(user) {
-			if (user.spaceId === spaceId) {
-				grouped.push(user);
-			}
-		});
-		const res: PaginatorApi<SpaceUser> = {
-			count: grouped.length,
-			results: grouped.slice(limit * page, limit * page + limit),
-		};
-		return of(res)
-			.pipe(delay(2000));
+	getUserList(spaceId: string, limit: number, page: number): Observable<Paging<SpaceUser>> {
+		return this.http.get<Paging<SpaceUser>>(this._apiUrl + "/user/list?start=" + limit*page + "&limit=" + limit + "&spaceId=" + spaceId).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		);
 	}
 
-	getLinkList(limit: number, page: number, spaceId: string = null): Observable<PaginatorApi<SpaceLink>> {
-		const grouped = spaceId ?
-			this.links.filter(item => item.spaceId === spaceId) :
-			this.links;
-		const res: PaginatorApi<SpaceLink> = {
-			count: grouped.length,
-			results: grouped.slice(limit * page, limit * page + limit),
-		};
-		return of(res)
-			.pipe(delay(2000));
+	getLinkList(limit: number, page: number, spaceId: string = null): Observable<Paging<SpaceLink>> {
+		let url = this._apiUrl + '/invitation/link?start=' + limit*page + '&limit=' + limit;
+		if (spaceId)
+			url += '&spaceId=' + spaceId;
+		return this.http.get<Paging<SpaceLink>>(url).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		);
 	}
 
-	isExists(key: string): Observable<boolean> {
-		return of(this.spaces.some(s => s.key === key))
-			.pipe(delay(2000));
+	isExists(key: string): Observable<any> {
+		return this.http.head(this._apiUrl + "/?key=" + key, { observe: 'response' })
+			.pipe(
+				src => this.requestWatcher.WatchFor(src),
+				catchError(error => {
+					if (error.status == 404)
+						return of(null);
+				}),
```

Comment:
Мы должны бросить ошибку дальше если это не 404
```suggestion
				catchError(err => {
					if (err.status == 404)
						return of(false);
					return throwError(() => err);
				}),
```

Reply:
Готово.